simtools/db/db_array_elements.py

```diff
@@ -0,0 +1,122 @@
+"""Retrieval of array elements from the database."""
+
+from functools import cache
+
+from pymongo import ASCENDING
+
+
+@cache
+def get_array_elements(db, model_version, collection):
+    """
+    Get all array element names and their design model for a given DB collection.
+
+    Uses the 'design_model' parameter to determine the design model of the array element.
+    Assumes that a design model is defined for every array element.
+
+    Parameters
+    ----------
+    db: DBHandler
+        Instance of the database handler
+    model_version: str
+       Model version.
+    collection: str
+        Database collection (e.g., telescopes, calibration_devices)
+
+    Returns
+    -------
+    dict
+        Dict with array element names found and their design model
+
+    Raises
+    ------
+    ValueError
+        If query for collection name not implemented.
+    KeyError
+        If array element entry in the database is incomplete.
+
+    """
+    db_collection = db.get_collection(db_name=None, collection_name=collection)
+
+    query = {"version": db.model_version(model_version)}
+    results = db_collection.find(query, {"instrument": 1, "value": 1, "parameter": 1}).sort(
+        "instrument", ASCENDING
+    )
+
+    _all_available_array_elements = {}
+    for doc in results:
+        try:
+            if doc["parameter"] == "design_model":
+                _all_available_array_elements[doc["instrument"]] = doc["value"]
+        except KeyError as exc:
+            raise KeyError("Incomplete array element entry in the database.") from exc
```

Comment:
Would it be worth logging or printing here which doc in results has the incomplete entry?

Reply:
Good point, added it.